nncf/common/pruning/operations.py

```diff
@@ -222,6 +223,42 @@ def mask_propagation(cls, node: NNCFNode, graph: NNCFGraph,
         result_mask = cls.generate_output_mask(node, graph, tensor_processor)
         node.data['output_mask'] = result_mask
 
+class SplitPruningOp(BasePruningOp):
+    @classmethod
+    def accept_pruned_input(cls, node: NNCFNode):
+        return True
+    
+    @classmethod
+    def generate_output_masks(cls, node: NNCFNode, graph: NNCFGraph,
+                              tensor_processor: Type[NNCFPruningBaseTensorProcessor]) -> Optional[NNCFTensor]:
+        """
+        Generate output mask
+        """
+        input_edges = graph.get_input_edges(node)
+        previous_nodes = [edge.from_node for edge in input_edges]
+        input_mask = previous_nodes[0].data['output_mask']
+
+        output_edges = graph.get_output_edges(node)
+        next_nodes = [edge.to_node for edge in output_edges]
+        output_shapes = [get_input_channels(node) for node in next_nodes]
+
+        chunk_size = node.layer_attributes.chunks
+        chunk_axis = node.layer_attributes.axis
+
+        result_masks = tensor_processor.split(input_mask, chunk_size, chunk_axis)
+        if len(set(output_shapes)) == 1:
+            for i, result_mask in enumerate(result_masks):
+                next_nodes[i].data['input_mask'] = result_mask
+        else:
+            idx = output_shapes.index(result_mask.shape[0])
+            next_nodes[idx].data['input_mask'] = result_mask
+        return result_masks
+
+    @classmethod
+    def mask_propagation(cls, node: NNCFNode, graph: NNCFGraph,
+                         tensor_processor: Type[NNCFPruningBaseTensorProcessor]) -> None:
+        result_mask = cls.generate_output_masks(node, graph, tensor_processor)       
+        node.data['output_mask'] = result_mask
```

Comment:
`output_mask` become list of masks instead of mask, this could lead to bugs in code

Reply:
It is already list, but to reduce confusion, modified it to add `s`.